planner/implementation/sort.go

```diff
@@ -37,3 +37,18 @@ func (impl *SortImpl) CalcCost(outCount float64, children ...memo.Implementation
 	impl.cost = sort.GetCost(cnt) + children[0].GetCost()
 	return impl.cost
 }
+
+// NominalSortImpl is the implementation of NominalSort.
+type NominalSortImpl struct {
+	baseImpl
+}
+
+// AttachChildren implements Implementation AttachChildren interface.
+func (impl *NominalSortImpl) AttachChildren(children ...memo.Implementation) memo.Implementation {
+	return children[0]
+}
+
+// NewNominalSortImpl creates a new NominalSort Implementation.
+func NewNominalSortImpl(sort *plannercore.NominalSort) *NominalSortImpl {
```

Comment:
Where do you eliminate the nominal sort?

Reply:
It is a little tricky... in `AttachChildren(...Implementation) Implementation`.